modules/core/src/v2/coins/eth.ts

```diff
@@ -518,7 +522,7 @@ export class Eth extends BaseCoin {
 
     const ethTxParams = {
       to: params.walletContractAddress,
-      nonce: params.signingKeyNonce,
+      nonce: params.signingKeyNonce || params.txPrebuild.halfSigned.backupKeyNonce,
```

Comment:
What if signingKeyNonce exists and == 0? Then we will use backupKeyNonce here

Reply:
This is a real bug. If we update to TypeScript 3.7+ (currently 3.5.2, should be no breaking changes) we can use [nullish coalescing](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-7.html#nullish-coalescing), which was introduced for exactly this reason

```suggestion
      nonce: params.signingKeyNonce ?? params.txPrebuild.halfSigned.backupKeyNonce,
```

Otherwise it should read

```suggestion
      nonce: params.signingKeyNonce !== undefined ? params.signingKeyNonce : params.txPrebuild.halfSigned.backupKeyNonce,
```

Reply:
Created a ticket for this issue: https://bitgoinc.atlassian.net/browse/BG-25181